Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/contests/contest-register/ContestRegister.tsx

```diff
@@ -39,33 +39,46 @@ const ContestRegister = () => {
         if (isLoading) {
             return;
         }
+        if (isRegisteredSuccessfully && !shouldConfirmParticipation && !requirePassword) {
+            navigate(`/contests/${contestId}/${participationType}`, { replace: true });
+        }
+    }, [ isLoading, isRegisteredSuccessfully, navigate, contestId, participationType, shouldConfirmParticipation, requirePassword ]);
 
-        if (!isRegisteredSuccessfully) {
+    // register user automatically if no password or modal confirmation is required
+    useEffect(() => {
+        if (isLoading || !data) {
+            return;
+        }
+        if (!requirePassword && !shouldConfirmParticipation && !isRegisteredSuccessfully) {
+            // eslint-disable-next-line promise/catch-or-return
             registerUserForContest({
                 id: Number(contestId),
-                isOfficial: participationType === ContestParticipationType.Compete,
+                isOfficial: participationType === 'compete',
```

Comment:
Why `ContestParticipationType.Compete` is removed and a string is introduced instead?

Reply:
fixed